### ISSUE TITLE
Proposal: Use storage backends for machine metadata

### ISSUE THREAD

**Issue by ekristen** [docker/machine#1408]
Instead of storing everything locally, it might be advantageous to use a storage backend like Consul or etc. This would allow easier sharing of machine data between multiple server admins. This would also align docker-machine with the likes of swarm, machine and others that are now supporting multiple key/value store systems for orchestration and discovery.

**Comment by nathanleclaire** [comment]
Indeed, this is why we made `Store` an interface (so there could be multiple implementations), but there are a few things which need to be ironed out first before it is really quite viable.

**Comment by ekristen** [comment]
Any movement on this idea at all? I'd really like to write one for Consul or even Vault for storing the private key and ssh keys, this would also make it easier to share with other members of my team.

**Comment by nathanleclaire** [comment]
Well, in the next "sprint" of stuff we want to re-work `libmachine` to make this kind of thing easier, so stay tuned.  That would include re-working `Store` to actually handle things like certs which are complected in the logic today.

**Comment by sukrit007** [comment]
+1

**Comment by anarcher** [comment]
+1
